### DISCUSSION TITLE
Provide a way to fix errors for rules without fixers

### DISCUSSION THREAD

**Discussion by gioragutt** [eslint/eslint#15023] [Ideas]
I'm in a repo using `yarn workspaces`. I often write code and let the IDE autocomplete me and auto-import as necessary.
However, often the IDE will use relative path imports and not use the package name import: `../../../modules/navigation` instead of `@my-repo/navigation`.

`eslint-plugin-import` contains a rule called `import/no-relative-packages`, which catches this kind of imports.

I know how to fix these kinds of issues, but AFAIK I don't have any way of extending rules/reacting to reports of other rules.

I currently have a custom rule which does that, but I also have to write the import analyzing code, which `eslint-plugin-import` already has, so it's a shame I cannot reuse "artifacts" of other rules.

**Comment by nzakas** [comment]
It sounds like you should contact the owners of eslint-plugin-import to ask them to expose the functionality you want to reuse.

**Reply by gioragutt** [reply]
In a way, you're right. But what if we ignore my specific use case and think about other cases where people want to automate fixing warnings/errors cause by eslint rules, in a way that's specific to their codebase?

**Reply by nzakas** [reply]
This cannot be solved in the abstract. Rules can be written in any number of ways.

**Reply by gioragutt** [reply]
I don't understand what you mean. I'm not saying that the rule should be changed.
I'm saying that you can add a mechanism to add fixers to other, existing rules.
Without altering the existing rule, but reacting to its output.

**Reply by nzakas** [reply]
I’m saying no that’s not possible.

**Reply by gioragutt** [reply]
Okay, so for the sake of discussion - why?
